test/e2e/chaos/networkchaos/misc.go

```diff
@@ -182,30 +182,41 @@ func makeNetworkDelayChaos(
 func probeNetworkCondition(c http.Client, peers []*corev1.Pod, ports []uint16) map[string][][]int {
 	result := make(map[string][][]int)
 
-	testDelay := func(from int, to int) int64 {
-		delay, err := testNetworkDelay(c, ports[from], peers[to].Status.PodIP)
-		framework.ExpectNoError(err, fmt.Sprintf(
-			"send request from %s to %s to test delay failed",
-			peers[from].Name, peers[to].Name,
-		))
-		return delay
+	testDelay := func(from int, to int) (int64, error) {
+		return testNetworkDelay(c, ports[from], peers[to].Status.PodIP)
 	}
 
 	for source := 0; source < len(peers); source++ {
 		for target := source + 1; target < len(peers); target++ {
```

Comment:
I think we could do even more further: how about making all the calls on `couldConnect()` parallel?

Reply:
but it may cause receive the response from other requests

Reply:
Yes, you are right. 🙈 We could not do this currently.